neural_speed/models/model_utils/pool.cpp

```diff
@@ -0,0 +1,94 @@
+//  Copyright (c) 2023 Intel Corporation
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+#include "models/model_utils/pool.h"
+
+serve_policy parse_serve_policy(const std::string& policy) {
+  if (policy == "fcfs") {
+    return serve_policy::FCFS;
+  } else {
+    fprintf(stderr, "Unexpected serve_policy %s!", policy.c_str());
+    return serve_policy::UNKNOWN;
+  }
+}
+
+// fcfs_pool
+bool fcfs_pool::add(sequence seq) {
+  context.emplace(seq);
+  return true;
+}
+
+bool fcfs_pool::pop(sequence* seq) {
+  if (empty()) {
+    fprintf(stderr, "%s: pool is empty.\n", __func__);
+    return false;
+  }
+  *seq = context.front();
+  context.pop();
+  return true;
+}
```

Comment:
why must return a bool instead of void function?

Reply:
just for unified actions and check each part in the `step` process.